rs/messaging/src/state_machine.rs

```diff
@@ -38,6 +40,7 @@ pub(crate) struct StateMachineImpl {
     scheduler: Box<dyn Scheduler<State = ReplicatedState>>,
     demux: Box<dyn Demux>,
     stream_builder: Box<dyn StreamBuilder>,
+    best_effort_message_memory_capacity: NumBytes,
```

Comment:
Because it's `StreamHandler` (i.e. induction) on the one hand; and Execution in general (via internal accounting plus `ReplicatedState::push_input(msg, subnet_available_memory)`) on the other; that enforce the guaranteed response message memory limit. And they both have it in some field or other.

Whereas the best-effort message memory limit is more porous (at least for now), so it's only enforced by `StateMachineImpl` at the end of every round.

Reply:
Thanks for the explanation. It does feel a bit weird that external components enforce the memory limits, but not something to change now.